### PR TITLE
Add shellcheck/ruff tests

### PR DESCRIPTION
**Pull request by nrobinaubertin**
- Adds a shellcheck and ruff tests
- Applies their recommendations

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -25,6 +25,34 @@ jobs:
...
           DJANGO_SECRET_KEY="testkey" python3 manage.py test
 
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run ShellCheck
+        run: |
+          shellcheck service/bin/* snap/hooks/*
```
Comment (arturo-seijas) [inline .github/workflows/tests.yaml:35]: Can you enclose there paths in double quotes?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -25,6 +25,34 @@ jobs:
           cd api
           DJANGO_SECRET_KEY="testkey" python3 manage.py test
 
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (arturo-seijas) [inline .github/workflows/tests.yaml:32]: Why `v2`? You are using `v3`in other places. I'd also pin to an exact version and let renovate do the updating

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -25,6 +25,34 @@ jobs:
...
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.x'
```
Comment (arturo-seijas) [inline .github/workflows/tests.yaml:46]: I'd pin here too

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (nrobinaubertin) [reply .github/workflows/tests.yaml:35]: `*` does not expand inside quotes

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (nrobinaubertin) [reply .github/workflows/tests.yaml]: Fixed and pinned to the latest version

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (nrobinaubertin) [reply .github/workflows/tests.yaml:46]: Fixed and pinned to the latest version

**Review by arturo-seijas — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Pull request merged**